core/node/metadata_calculator/src/updater.rs

```diff
@@ -152,10 +152,6 @@ impl TreeUpdater {
             // right away without having to implement dedicated code.
 
             if let Some(object_key) = &object_key {
-                storage
-                    .tee_verifier_input_producer_dal()
-                    .create_tee_verifier_input_producer_job(l1_batch_number)
-                    .await?;
                 // Save the proof generation details to Postgres
                 storage
```

Comment:
I know it's not your code, but do you happen to know why we have two separate calls here? Why not delete the `.save_merkle_paths_artifacts_metadata()` call and save the Merkle paths within the `.insert_proof_generation_details(l1_batch_number, object_key)` call instead? Just curious if there's a good reason for it. 🤔

Reply:
Ah, it looks like @Artemka374 fixed a race condition [here](https://github.com/matter-labs/zksync-era/pull/2405). Honestly, I don't fully understand it just by reading the code, but it seems necessary.